build-scripts/runtime_lib.cmake

```diff
@@ -105,8 +105,7 @@ endif ()
 
 ####################### Common sources #######################
 if (NOT MSVC)
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -ffunction-sections -fdata-sections \
-                                         -Wall -Wno-unused-parameter -Wno-pedantic")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -ffunction-sections -fdata-sections -Werror")
```

Comment:
Will it be too strict for all platforms and targets? For example, #991 reports a lot of warnings for thumb when compiling with Wcast-align. Shall we only add '-Werror' for some common used platforms and targets, e.g. Linux/MacOS/Windows and x86-64?

Reply:
Yes we can consider it, in fact I'm working with clang on my local machine to cleanup these warnings, may not works on all platforms.

I'll try to let this PR to pass CI first.

Reply:
I agree with @wenyongh , had better apply this strict check for some specific platforms. 

I remember that there are also a lot of warnings on Windows(MSVC).

Reply:
It's not a problem since the best way is to use -Werror for all target, if we can't achive this, as a fallback we can apply it for partial target only.